WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostWebViewCachingFragment.java

```diff
@@ -0,0 +1,83 @@
+package org.wordpress.android.ui.reader;
+
+
+import android.app.Fragment;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+import org.wordpress.android.datasets.ReaderPostTable;
+import org.wordpress.android.models.ReaderPost;
+import org.wordpress.android.ui.reader.views.ReaderWebView;
+import org.wordpress.android.util.NetworkUtils;
+
+/**
+ * Fragment responsible for caching post content into WebView.
+ * Caching happens on UI thread, so any configuration change will restart it from scratch.
+ */
+public class ReaderPostWebViewCachingFragment extends Fragment {
+    private static final String ARG_BLOG_ID = "blog_id";
+    private static final String ARG_POST_ID = "post_id";
+
+    private long mBlogId;
+    private long mPostId;
+
+    public static ReaderPostWebViewCachingFragment newInstance(long blogId, long postId) {
+        ReaderPostWebViewCachingFragment fragment = new ReaderPostWebViewCachingFragment();
+        Bundle args = new Bundle();
+        args.putLong(ARG_BLOG_ID, blogId);
+        args.putLong(ARG_POST_ID, postId);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setRetainInstance(true);
```

Comment:
What are some benefits of using retained fragment in this case?

Reply:
Oops. I used it with initial implementation and forgot to remove. Good catch!